R/simulate.r

```diff
@@ -206,30 +193,20 @@ simulate_chains <- function(index_cases,
   }
   # next, simulate n trees
   while (length(sim) > 0 && susc_pop > 0) {
-    # sample next generation of offspring
-    next_gen <- do.call(
-      offspring_dist,
-      c(
-        list(n = sum(n_offspring[sim])),
-        pars
+    # simulate the next possible offspring
+    next_gen <- .sample_possible_offspring(
+      offspring_func = offspring_dist,
+      offspring_func_pars = pars,
+      n_offspring = n_offspring,
+      chains = sim
       )
+    # from all possible offspring, get those that could be infected
+    next_gen <- .get_infectible_offspring(
```

Comment:
```suggestion
    next_gen <- .get_susceptible_offspring(
```
to be more in line with the rest of the terminology